tfutils/model.py

```diff
@@ -1,254 +1,216 @@
+from __future__ import absolute_import, division, print_function
+from collections import OrderedDict
+
+import numpy as np
 import tensorflow as tf
 
 
 class ConvNet(object):
     """Basic implementation of ConvNet class compatible with tfutils.
     """
-    def __init__(self, train=True, seed=0, cfg_initial=None):
-        self.train = train
+
+    def __init__(self, seed=None, **kwargs):
         self.seed = seed
-        self.conv_counter = 0
-        self.pool_counter = 0
-        self.fc_counter = 0
-        self.norm_counter = 0
-        self.global_counter = 0
-        self.parameters = []
-        self.architecture = {}
-
-    def conv(self, 
-             in_layer,
+        self.output = None
+        self._params = OrderedDict()
+
+    @property
+    def params(self):
+        return self._params
+
+    @params.setter
+    def params(self, value):
+        name = tf.get_variable_scope().name
+        if name not in self._params:
+            self._params[name] = OrderedDict()
+        self._params[name][value['type']] = value
+
+    @property
+    def graph(self):
+        return tf.get_default_graph().as_graph_def()
+
+    def initializer(self, kind='xavier', stddev=.1):
+        if kind == 'xavier':
+            init = tf.contrib.layers.initializers.xavier_initializer(seed=self.seed)
+        elif kind == 'trunc_norm':
+            init = tf.truncated_normal_initializer(mean=0, stddev=stddev, seed=self.seed)
+        else:
+            raise ValueError('Please provide an appropriate initialization '
+                             'method: xavier or trunc_norm')
+        return init
+
+    def conv(self,
              out_shape,
              ksize=3,
              stride=1,
              padding='SAME',
+             init='xavier',
              stddev=.01,
              bias=1,
-             name=None):
+             activation='relu',
+             weight_decay=None,
+             in_layer=None):
+        if in_layer is None: in_layer = self.output
```

Comment:
I prefer multi-line if's to these single-line statements, but am not wedded to it.